service/src/main/java/uk/gov/hmcts/reform/fpl/service/CreateC21OrderService.java

```diff
@@ -96,4 +106,13 @@ private String getCourtName(String courtName) {
                     .formatLocalDateToString(child.getDateOfBirth(), FormatStyle.LONG) : ""))
             .collect(toList());
     }
+
+    private String mostRecentUploadedC21DocumentUrl(final List<Element<C21Order>> c21Orders) {
+        return getLast(c21Orders.stream()
```

Comment:
Hmm I am not sure if I like the approach here, getLast relies on the order provided to us by CCD, I think it's bad practice and makes code a lot more difficult in future to change than required. I know it's probably correct in this instance (and it will be 99% of the time in future), but maybe we should return really what mostRecent means? order by date and take the latest one?

Reply:
Probbaly before collecting mapping would be a bit nicer :)

Reply:
@joez-f what do you think about the above?

Reply:
I like sorting by orderDate and then taking the latest one - good to do it explicitly (although I do agree that in this instance I don't see a way that the list of c21Orders could ever be out of order. so not a huge problem IMO, but let's explicitly return mostRecent if it's not too difficult)

Reply:
I picked this option as the _orderDate_ is currently stored in string